### PR TITLE
Documentation porting and versioning

### PR DESCRIPTION
**Pull request by giobart**
This PR includes a porting of the original documentation.
image

To do so, it provides a releases selector.
image

Old wiki releases show an alert banner:
image

The original Accordion documentation is part of an archive directory
image

All the archived versions of the wiki can be moved there.

To include a new version of the documentation to the release selector the file `layout/partials/versions.html` must be edited. E.g.:
```
<li><a class="dropdown-item current" aria-current="true" href="/docs/getting-started/oak-environment/high-level-setup-overview/">Latest: Bass (v{{ site.Params.doks.docsVersion }}) &nbsp;&nbsp;</a></li>
<li><hr class="dropdown-divider"></li>
<li><a class="dropdown-item" href="/archive/0.4.301/docs/getstarted/get-started-cluster/">Accordion v0.4.301</a></li>

```

### CODE REVIEW

**Comment by Malyuk-A** [comment]
When I click on the latest link: 
image

I get to this page: docs/

Is this as intended?

image

**Comment by Malyuk-A** [comment]
I have never seen GitHub struggle so much displaying the file changes - this PR has now several times broken my github page  - 5k+ FILE changes - massive bloat

I think I know what the issue is:

image

We have a folder called fontawesome.
I have worked with it once.

IMO there are ~2 ways to work with it:
Online: easy online include - no need to store any files locally - see https://www.youtube.com/watch?v=sJDyEB1uWCw
Offline: Similar to what we do - but we could cut out everything that we do not need and reduce this by ~99%

AFAIK - we only use fontawesome for a SINGLE icon - a github icon: 
image

We could easily replace this icon with a SVG without fontawesome at all - see https://www.youtube.com/watch?v=or7amkb0Pk8

**Comment by giobart** [comment]
> > "This PR includes a porting of the original documentation."
> 
> Why only a portion? And what do you mean by portion - what bits are not added to this archive?

a porting not a portion

File: layouts/archive/single.html [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+						{{ partial "main/breadcrumb" . -}}
+						<li class="breadcrumb-item active" aria-current="page">{{ .Title }}</li>
+					</ol>
+				</nav>
+			{{ end }}
+			
+			<h1>{{ .Title }}</h1>
+
+			<div class="grey-box">
+                <span class="alert-icon">&#9888;</span> This documentation refers to a version of Oakestra which is not the latest. Please refer to <a href="/docs">latest</a>.
```
Comment (Mjaethers) [inline layouts/archive/single.html]:
```suggestion
                <span class="alert-icon">&#9888;</span> This page documents a previous version of Oakestra. Please refer to the <a href="/docs/getting-started/oak-environment/high-level-setup-overview/">latest docs</a> for a current version.
```
I think the phrasing here is a bit off

**Review by Mjaethers — COMMENTED** [review]
I agree with @Malyuk-A, this is a bit hard to review on account of the number of files, but in general it looks good

**Comment by giobart** [comment]
TODO:

- Move publications inside Docs
- Remove Blog
- Remove github link in landing page 
- change `archive` to `version`

**Comment by giobart** [comment]
# Updates:

#### Changed Oakestra text with bw oakestra logo and added publications section:

image
image

#### The "publication" section is an auto-generated list of publications
image

The list is generated from the publication files under the publications folder 
image

Each publication file simply contains the pub info as follows
```
---
title: "Oakestra: A Lightweight Hierarchical Orchestration Framework for Edge Computing"
authors: "Giovanni Bartolomeo, Mehdi Yosofie, Simon Bäurle, Oliver Haluszczynski, Nitinder Mohan, Jörg Ott"
conference: "USENIX Annual Technical Conference (ATC), 2023"
img: "/pubs/Oakestra-ATC2023.png"
badges: ["/img/usenix-badges/usenixbadges-available.png","/img/usenix-badges/usenixbadges-functional.png","/img/usenix-badges/usenixbadges-reproduced.png"]
pdf: "/pubs/Oakestra-ATC2023.pdf"
doi: "https://www.usenix.org/conference/atc23/presentation/bartolomeo"
summary: ""
draft: false
weight: 204
toc: false
---
```


#### Cleaned landing page

image


#### Added simple welcome blogpost in the blog 

image

#### Removed the garbage thousands of fontawesome files

#### Added discord handle in the navbar

image


@Malyuk-A @nitindermohan  @Mjaethers lmk if everything is in order so we can push it.

**Comment by Malyuk-A** [comment]
Nitpick / Followup: 

image

How hard is it to center the privacy policy - currently, it is off-center 😅

**Comment by Malyuk-A** [comment]
Nitpick / Followup:

The Discord link does not seem to work:

image

**Comment by giobart** [comment]
> Nitpick / Followup:
> 
> The Discord link does not seem to work:
> 
> image

That's the one from previous docs, I guess we need to generate a new one then? Is the one in oak website working for you? It's weird but it works for me...

**Comment by Malyuk-A** [comment]
The publications are not displayed properly on my end:

image
image


I had a quick look and tried a few quick things with no success - If you want me to investigate further, let me know (this happens for me on Brave and Chrome)

Same goes for the actual paper links:

image

The last DOI points to nothing:
image

**Comment by giobart** [comment]
> The publications are not displayed properly on my end:

Can you quickly try to stop the server, remove the 'public/' folder and re-start it?

**Comment by Malyuk-A** [comment]
> That's the one from previous docs, I guess we need to generate a new one then? Is the one in oak website working for you? It's weird but it works for me...

OK nvm - link is fine when I open it in incognito mode - when I do this in my regular browser mode it does not - I assume this is because I also have my browser linked to discord - so it knows that I am already a member and it goes stupid - that is a discord problem not ours - all good
image

**Comment by Malyuk-A** [comment]
> > The publications are not displayed properly on my end:
> 
> Can you quickly try to stop the server, remove the 'public/' folder and re-start it?

I did - same issue

P.S: Why does github not support threads in PR comments - in gitlab this is soo much cleaner and easier to follow what is going on :(

**Comment by giobart** [comment]
> > > The publications are not displayed properly on my end:
> > 
> > 
> > Can you quickly try to stop the server, remove the 'public/' folder and re-start it?
> 
> I did - same issue

Can you check with latest push?

> P.S: Why does github not support threads in PR comments - in gitlab this is soo much cleaner and easier to follow what is going on :(

You're totally right! :(

**Comment by Malyuk-A** [comment]
>Can you check with latest push?
Sadly still not there - If you want me to I can Investigate this tomorrow

**Comment by Malyuk-A** [comment]
I think this is up for debate:

Nitpick/Personal-taste:

I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me

image

Here are a couple of other examples:

image
https://kubernetes.io/docs/home/

image
https://unikraft.org/


Personally I would do this: 
GetStarted - Releases - GitHub - discord - search - dark/light

**Comment by giobart** [comment]
> > Sadly still not there - If you want me to I can Investigate this tomorrow

That's absurd D: I wonder what the heck is going on with the images. Tho its just the images right? How do you run the project?

**Comment by giobart** [comment]
> I think this is up for debate:
> 
> Nitpick/Personal-taste:
> 
> I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me
> 
> Personally I would do this: GetStarted - Releases - GitHub - discord - search - dark/light

Yup the order can be changed, this is i jsut the default for the theme. We should talk about it.

**Comment by Mjaethers** [comment]
Personal taste:
I think the blue logo looks a lot nicer than the black one and also works well in light mode, I think we should always use it.
image
image

**Comment by Mjaethers** [comment]
The prev and next cards on the Accordion docs are all over the place. Is this something worth spending time on fixing right now?

**Comment by giobart** [comment]
> I think this is up for debate:
> 
> Nitpick/Personal-taste:
> 
> I personally have a strange feeling about the current layout/order of the header elements on the right (icons, release, search, etc.) - it feels off to me
>
> Personally I would do this: GetStarted - Releases - GitHub - discord - search - dark/light

Based on discussion, I'll create a draft with the following features:
- releases not showing up but shows only while inside Docs section. 
- Left side: Docs Blog Research
- Right side: Social - search - light/dark - get started
- get started gets replaced with releases when inside Docs.
- remove old docs from search
- Fix font changes

**Comment by nitindermohan** [comment]
> Personal taste: I think the blue logo looks a lot nicer than the black one and also works well in light mode, I think we should always use it. image image

I second this!

**Comment by Malyuk-A** [comment]
I think the following files can be deleted:

image

- assets/img/avatar_hu7fc94d4596db6a879669a889061e6268_294933_300x0_resize_box_3.png
- content/version/0.4.301/docs/oakestra/node-engine.md
- content/version/0.4.301/docs/oakestra/motivations.md

The following files could maybe be deleted but I am not certain (e.g. I could not find where they get used)
- [assets/dashboard/logo.png](https://github.com/nitindermohan/oak-new-docs/pull/15/files#diff-6d5e5ab83e5227972e24f359bfe7f16d3dad8590511440affce132a9277efc57)
- [assets/img/avatar.png](https://github.com/nitindermohan/oak-new-docs/pull/15/files#diff-994f3eb21fe0aa94bc694529ffc361fba2f566cb2efd31bdab10e9e30c9e82f7)
- assets/network/resolution-seq.png
- content/version/0.4.301/docs/oakestra/README-old.md
- static/css/font-awesome.css
- static/css/font-awesome.min.css
- static/fonts/FontAwesome.otf
- static/fonts/fontawesome-webfont.eot
- static/fonts/fontawesome-webfont.svg
- static/fonts/fontawesome-webfont.ttf
- static/fonts/fontawesome-webfont.woff
- static/fonts/fontawesome-webfont.woff2

Could it be the case that the entire "oakestra" folder in the legacy docs are not used and we could delete it?
image

**Comment by Malyuk-A** [comment]
Besides that - I think everything else looks good
I hope that we can merge this PR quickly because it introduces plenty of (isolated) changes - and many nitpicks we (especially) I have/had can also be easily resolved in smaller targeted follow-up PRs. 🙂

**Comment by giobart** [comment]
image
image

- Fixed logo (now just small icon)
- Changed Icons order
- Releases hidden in home page
- Hidden getstarted inside pages != landing page
- uniform fonts
- removed old docs from search index 
- rename publication sections into research
- fixed releases toggle, now showing selection `latest` only when actually in latest docs 
- removed some garbage

File: content/research/_index.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+---
+title: "Research" # Figure out why title not working
```
Comment (Malyuk-A) [inline content/research/_index.md]: Has this been resolved?  - if so the comment is outdated

**Review by Malyuk-A — APPROVED** [review]
Now it looks a lot cleaner and more professional - Truly great job Giovanni!

There are a few very tiny bits like unnecessary empty lines or commented bits here and there in some HTML files - but to be frank - I would merge this ASAP and if need be tidy up / polish some other time

Ups - I thought I already send this out nmv - now ^^

File: content/research/_index.md (hunk 2 above)
Comment (giobart) [reply content/research/_index.md]:
```suggestion
title: "Research"
```

**Comment by nitindermohan** [comment]
A few nitpicks from my end:

1. Subtitle font on front page is too small
Screenshot 2024-12-10 at 17 18 21

2. Footer has a weird spacing at the bottom of the page
Screenshot 2024-12-10 at 17 18 32

3. Release button does not show anything on click
Screenshot 2024-12-10 at 17 18 42

4. DOI button shows up weirdly for publications (sometime in same line as conf name, sometime in next line)
Screenshot 2024-12-10 at 17 19 09
Screenshot 2024-12-10 at 17 19 20

**Comment by giobart** [comment]
> A few nitpicks from my end:
> 
> 1. Subtitle font on front page is too small

Increased font size

> 2. Footer has a weird spacing at the bottom of the page
> 
> 3. Release button does not show anything on click

These were big ones to solve. At first I was not able to replicate the issue, then I noticed that it only occurs on some browsers. After some tinkering I ended up manually adding bootstrap as a module and loading the bootstrap js partials manually in the navbar. Now it looks like it's working.  Please test it once more @nitindermohan 

>  4. DOI button shows up weirdly for publications (sometime in same line as conf name, sometime in next line)

I think I fixed it. But on the whitepaper the conference line remains blank.

**Comment by nitindermohan** [comment]
Getting build error on the branch 
```
Error: command error: failed to load modules: failed to download modules: binary with name "go" not found in PATH
```
It is caused by the go.mod and go.sum files that were pushed in commit https://github.com/nitindermohan/oak-new-docs/pull/15/commits/58605296eb6328fcdc75d4caa8185a9ab99f2fa3. Maybe artifacts of the other website as they still refer to another documentation repo.

EDIT: Works after installing `golang` in the machine. @giobart will add new documentation setup instructions with the GoLang dependency.

**Comment by nitindermohan** [comment]
The PR looks good now. Merging it! Thanks all for your contributions :)

**Pull request merged**